buildSrc/src/main/java/org/springframework/security/CheckExpectedBranchVersionPlugin.java

```diff
@@ -36,38 +44,67 @@ public void apply(Project project) {
 		TaskProvider<CheckExpectedBranchVersionTask> checkExpectedBranchVersionTask = project.getTasks().register("checkExpectedBranchVersion", CheckExpectedBranchVersionTask.class, (task) -> {
 			task.setGroup("Build");
 			task.setDescription("Check if the project version matches the branch version");
+			task.onlyIf("skipCheckExpectedBranchVersion property is false or not present", CheckExpectedBranchVersionPlugin::onlyIfCheck);
+			task.getVersion().convention(project.provider(() -> project.getVersion().toString()));
+			task.getBranchName().convention(project.getProviders().exec(execSpec -> execSpec.setCommandLine("git", "symbolic-ref", "--short", "HEAD")).getStandardOutput().getAsText());
+			task.getOutputFile().convention(project.getLayout().getBuildDirectory().file("check-expected-branch-version"));
 		});
 		project.getTasks().named(JavaBasePlugin.CHECK_TASK_NAME, checkTask -> checkTask.dependsOn(checkExpectedBranchVersionTask));
 	}
 
-	public static class CheckExpectedBranchVersionTask extends DefaultTask {
+	/**
+	 * <p>If the project is configured with property <code>-PskipCheckExpectedBranchVersion</code> or set equal to any value other than <code>false</code> then the task will be skipped.</p>
+	 * <p>If the project is configured with property <code>-PskipCheckExpectedBranchVersion=false</code> or the property is absent then the task will be executed.</p>
+	 * @param task context to perform onlyIf check in.
+	 * @return <code>true</code> only if the <code>skipCheckExpectedBranchVersion</code> property is present or present with any value other than <code>false</code>.
+	 */
+	private static boolean onlyIfCheck(Task task) {
+		return task.getProject().getProviders().gradleProperty("skipCheckExpectedBranchVersion")
+				// if the property is not present then evaluate the property with a fallback value of false
+				.orElse("false")
+				// if the property value is equal to false do not skip
+				.map("false"::equalsIgnoreCase)
+				// if the property value is anything else skip running the task.
+				.get();
+	}
```

Comment:
I would drop the JavaDoc, comments, and rename the method. The logic is concise and straightforward enough that I think the comments distract from how simple it is.

```suggestion
	private static boolean skipPropertyFalseOrNotPresent(Task task) {
		return task.getProject().getProviders().gradleProperty("skipCheckExpectedBranchVersion")
				.orElse("false")
				.map("false"::equalsIgnoreCase)
				.get();
	}
```

Reply:
BTW, I don't think I've ever seen a method reference off of a string literal like that. Nice.